### PR TITLE
feat: camunda exporters support the removal of a member from a role

### LINKED ISSUE #24582: Camunda exporters support the removal of a member from a role
### Description 
To be able to keep data consistent between our Zeebe stores and Elasticsearch, we need to process `EntityRemoved` (member) events.

Important hints:
- In this [PR](https://github.com/camunda/camunda/pull/23864/files#diff-e538348bc6e8511b1303287000789b22891f06b226577f4cec3d2f80a49e8257R18) a new `RoleEntity` with a field `AssignedMemberKeys` is added. The set in Elasticsearch is complete and contains all assigned members.
- The `EntityRemoved` intent/event for a [RoleRecord](https://github.com/camunda/camunda/blob/main/zeebe/protocol-impl/src/main/java/io/camunda/zeebe/protocol/impl/record/value/authorization/RoleRecord.java#L22) contains a single key to be removed to the list, not the complete list. The handler must use a method that removes this single value from the field in ES.

### Acceptance Criteria
- [ ] A `RoleEntityRemovedHandler` exists [here](https://github.com/camunda/camunda/tree/main/zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers)
- [ ] The `RoleEntityAddedHandler` removed the single key from the `assignedMemberKeys` stored in ES
- [ ] A unit test is added to check behaviours - hint: [here](https://github.com/camunda/camunda/blob/main/zeebe/exporters/camunda-exporter/src/test/java/io/camunda/exporter/handlers/UserDeletedHandlerTest.java)

### PR DESCRIPTION
**Pull request by matthewBearCamunda**
## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #24582

### CODE REVIEW

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/role/RoleEntityRemovedHandler.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+import io.camunda.zeebe.protocol.record.intent.Intent;
+import io.camunda.zeebe.protocol.record.intent.RoleIntent;
+import io.camunda.zeebe.protocol.record.value.RoleRecordValue;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class RoleEntityRemovedHandler implements ExportHandler<RoleEntity, RoleRecordValue> {
```
Comment (emulbreh) [inline zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/role/RoleEntityRemovedHandler.java:26]: I believe we should convert assigned members to child documents as proposed in https://github.com/camunda/camunda/pull/24782 for groups.

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/role/RoleEntityRemovedHandler.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  }
+
+  @Override
+  public RoleEntity createNewEntity(final String id) {
+    return new RoleEntity().setId(id);
+  }
+
+  @Override
+  public void updateEntity(final Record<RoleRecordValue> record, final RoleEntity entity) {
+    final Set<Long> updatedAssignedMemberKeys = new HashSet<>(entity.getAssignedMemberKeys());
```
Comment (emulbreh) [inline zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/role/RoleEntityRemovedHandler.java:62]: This doesn't quite work as the entity isn't read from ES. I ran into the same problem when I worked on this in https://github.com/camunda/camunda/pull/24609.

**Comment by matthewBearCamunda** [comment]
Putting this PR on hold until https://github.com/camunda/camunda/pull/24782 is complete. I'll rework this PR to align with the changes with that once its complete

**Comment by matthewBearCamunda** [comment]
Closing in favor of https://github.com/camunda/camunda/pull/24609

**Pull request closed without merging**